### PR TITLE
[SPARK-45515][CORE][SQL][FOLLOWUP] Use enhanced switch expressions to replace the regular switch statement

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR follows up https://github.com/apache/spark/pull/43349.

This pr also does not include parts of the hive and hive-thriftserver module.

### Why are the changes needed?
Please see https://github.com/apache/spark/pull/43349.

### Does this PR introduce _any_ user-facing change?
'No'.

### How was this patch tested?
GA

### Was this patch authored or co-authored using generative AI tooling?
'No'.

### CODE REVIEW

**Comment by beliefer** [comment]
The GA failure is unrelated.

**Comment by beliefer** [comment]
@srowen @LuciferYang Please help me to review this one.

File: common/network-common/src/main/java/org/apache/spark/network/util/DBProvider.java [hunk 1]
```diff
@@ -38,17 +38,18 @@ public static DB initDB(
...
+          }
+          case ROCKSDB -> {
             org.rocksdb.RocksDB rocksDB = RocksDBProvider.initRockDB(dbFile, version, mapper);
-            return rocksDB != null ? new RocksDB(rocksDB) : null;
-          default:
-            throw new IllegalArgumentException("Unsupported DBBackend: " + dbBackend);
-        }
+            yield rocksDB != null ? new RocksDB(rocksDB) : null;
+          }
+          default -> throw new IllegalArgumentException("Unsupported DBBackend: " + dbBackend);
```
Comment (LuciferYang) [inline common/network-common/src/main/java/org/apache/spark/network/util/DBProvider.java:51]: How about

```scala
return switch (dbBackend) {
          case LEVELDB -> {
            logger.warn("The LEVELDB is deprecated. Please use ROCKSDB instead.");
            yield new LevelDB(LevelDBProvider.initLevelDB(dbFile, version, mapper));
          }
          case ROCKSDB -> new RocksDB(RocksDBProvider.initRockDB(dbFile, version, mapper));
        };
```
`dbBackend` is an enumeration type, the default case is not necessary.

File: launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java [hunk 2]
```diff
@@ -143,59 +143,61 @@ static List<String> parseOptionString(String s) {
         escapeNext = false;
       } else if (inOpt) {
         switch (c) {
-        case '\\':
-          if (inSingleQuote) {
-            opt.appendCodePoint(c);
-          } else {
-            escapeNext = true;
+          case '\\' -> {
```
Comment (LuciferYang) [inline launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java:146]: For this particular case, I personally believe that `Pattern Matching for switch` does not demonstrate its advantages.

File: sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetVectorUpdaterFactory.java [hunk 3]
```diff
@@ -181,9 +181,8 @@ public ParquetVectorUpdater getUpdater(ColumnDescriptor descriptor, DataType spa
         } else if (sparkType == DataTypes.BinaryType) {
           return new FixedLenByteArrayUpdater(arrayLen);
         }
-        break;
-      default:
-        break;
+      }
+      default -> {}
```
Comment (LuciferYang) [inline sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetVectorUpdaterFactory.java:185]: Just to confirm, can we move 

```java
throw constructConvertNotSupportedException(descriptor, sparkType);
``` 

from line 190 into the default branch?

File: sql/hive/src/test/java/org/apache/spark/sql/hive/test/Complex.java [hunk 4]
```diff
@@ -84,22 +84,15 @@ public enum _Fields implements org.apache.thrift.TFieldIdEnum {
      * Find the _Fields constant that matches fieldId, or null if its not found.
      */
     public static _Fields findByThriftId(int fieldId) {
```
Comment (LuciferYang) [inline sql/hive/src/test/java/org/apache/spark/sql/hive/test/Complex.java:86]: this file copy from Hive 0.13, let's leaving it as it is.

File: common/network-common/src/main/java/org/apache/spark/network/util/DBProvider.java (hunk 1 above)
Comment (LuciferYang) [reply common/network-common/src/main/java/org/apache/spark/network/util/DBProvider.java:51]: Please ignore this comments

File: launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java (hunk 2 above)
Comment (beliefer) [reply launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java:146]: Yes. This case can't be simplified. But it is safer than before.

File: common/network-common/src/main/java/org/apache/spark/network/util/DBProvider.java (hunk 1 above)
Comment (beliefer) [reply common/network-common/src/main/java/org/apache/spark/network/util/DBProvider.java:51]: We can remove the `default` here.

File: sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetVectorUpdaterFactory.java (hunk 3 above)
Comment (beliefer) [reply sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetVectorUpdaterFactory.java:185]: Good idea.

Comment (LuciferYang) [reply sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetVectorUpdaterFactory.java:185]: Got

File: launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java (hunk 2 above)
Comment (LuciferYang) [reply launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java:146]: OK

File: common/network-common/src/main/java/org/apache/spark/network/util/DBProvider.java (hunk 1 above)
Comment (LuciferYang) [reply common/network-common/src/main/java/org/apache/spark/network/util/DBProvider.java:51]: OK ~

**Comment by beliefer** [comment]
cc @dongjoon-hyun

**Review by dongjoon-hyun — APPROVED** [review]

File: sql/hive/src/test/java/org/apache/spark/sql/hive/test/Complex.java [hunk 5]
```diff
@@ -441,79 +441,79 @@ public void setMStringStringIsSet(boolean value) {
...
   public void setFieldValue(_Fields field, Object value) {
     switch (field) {
-    case AINT:
-      if (value == null) {
-        unsetAint();
-      } else {
-        setAint((Integer)value);
-      }
-      break;
+      case AINT:
```
[the remark this answers is not in the document]
Comment (dongjoon-hyun) [reply sql/hive/src/test/java/org/apache/spark/sql/hive/test/Complex.java:444]: Oh.. I thought this file is excluded according to your previous comment, https://github.com/apache/spark/pull/44183#discussion_r1416754836 .

[the remark this answers is not in the document]
Comment (dongjoon-hyun) [reply sql/hive/src/test/java/org/apache/spark/sql/hive/test/Complex.java:444]: Ya, there should be no change (including whitespace) on this file.

**Review by LuciferYang — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by beliefer** [comment]
The GA failure is unrelated.
Merged to Master.
@dongjoon-hyun @LuciferYang Thank you!

**Pull request closed without merging**